app/form/page.tsx

```diff
@@ -1,45 +1,57 @@
-"use client";
-import { useState, FormEvent, useEffect, ChangeEvent } from "react";
+'use client';
+import { useState, useEffect, FormEvent, ChangeEvent } from "react";
 import { useRouter } from "next/navigation";
 import { ZodError } from "zod";
 import formSchema from "../api/schema/schema";
 import Input from "../components/formInputs/Input";
-import { FormData } from "../components/formInputs/Input";
 import Image from "next/image";
 
 const ChatBotForm = () => {
-  const [formData, setFormData] = useState<FormData>({
+  const [formData, setFormData] = useState({
     name: "",
-    mobileNumber: "", 
+    mobileNumber: "",
     physicsMarks: 0,
     chemistryMarks: 0,
-    mathsMarks: 0,
+    rank: 0,
     course: "",
     community: "",
   });
 
   const [errors, setErrors] = useState<Record<string, string>>({});
   const [isLoading, setIsLoading] = useState(false);
-  const [cutoffMarks, setCutoffMarks] = useState<number | null>(null);
+  const [searchQuery, setSearchQuery] = useState("");
+  const [filteredColleges, setFilteredColleges] = useState<string[]>([]);
+  const [showDropdown, setShowDropdown] = useState(false); // State for dropdown visibility
   const router = useRouter();
 
-  useEffect(() => {
-    const calculatecutoff = () => {
-      const maths = formData.mathsMarks;
-      const physics = formData.physicsMarks / 2;
-      const chemistry = formData.chemistryMarks / 2;
+  const colleges = [
```

Comment:
Dont change in existing page, try to create new page and input form. As we need to reuse the other page as well.